src/test/java/it/gov/pagopa/idpay/transactions/repository/RewardTransactionSpecificRepositoryTest.java

```diff
@@ -0,0 +1,131 @@
+package it.gov.pagopa.idpay.transactions.repository;
+
+import it.gov.pagopa.idpay.transactions.BaseIntegrationTest;
+import it.gov.pagopa.idpay.transactions.model.RewardTransaction;
+import it.gov.pagopa.idpay.transactions.test.fakers.RewardTransactionFaker;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import reactor.core.publisher.Flux;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@DirtiesContext
+class RewardTransactionSpecificRepositoryTest extends BaseIntegrationTest {
+    //TODO pageable add null
+
+    @Autowired
+    protected RewardTransactionRepository rewardTransactionRepository;
+
+    @Autowired
+    private RewardTransactionSpecificRepositoryImpl rewardTransactionSpecificRepository;
+
+    private RewardTransaction rt;
+
+    @BeforeEach
+    void setUp(){
+        LocalDateTime date = LocalDateTime.of(2021, 9, 6, 17, 30, 25);
+        BigDecimal amount = new BigDecimal("30.00");
+        rt = RewardTransactionFaker.mockInstanceBuilder(1)
+                .id("id_prova")
+                .idTrxIssuer("IDTRXISSUER1")
+                .trxDate(date)
+                .amount(amount).build();
+        rewardTransactionRepository.save(rt).block();
+    }
+
+    @AfterEach
+    void clearData(){
+        rewardTransactionRepository.deleteById("id_prova");
+    }
+
+    @Test
+    void findByIdTrxIssuer() {
+        Flux<RewardTransaction> resultTrxIssuer = rewardTransactionSpecificRepository.findByIdTrxIssuer(rt.getIdTrxIssuer(),null ,null,null, null, null);
+        Assertions.assertNotNull(resultTrxIssuer);
+        Assertions.assertEquals(1, resultTrxIssuer.count().block());
+        RewardTransaction rtResultTrxIssuer = resultTrxIssuer.toStream().findFirst().orElse(null);
```

Comment:
by doing this, you are executing 2 time the reactive pipeline, thus executing two queries!
let's just execute it once, calling the block and storing the List, next you could test the List content.

Make this change also below